### PR TITLE
Add com.github.afrantzis.Bless

### PR DESCRIPTION
**Pull request by FakeShemp**
https://github.com/afrantzis/bless

### CODE REVIEW

**Comment by FakeShemp** [comment]
bot, build com.github.afrantzis.Bless

**Comment by FakeShemp** [comment]
There is only an 48x48 icon available for this app.

**Comment by bilelmoussaoui** [comment]
> There is only an 48x48 icon available for this app.

You can upscale it for now and maybe ask upstream if they can provide one in the future.

**Comment by FakeShemp** [comment]
I've added an upscaled version of the icon, but upstream's email bounces, so it's hard to contact them about Bless at the moment.

**Comment by FakeShemp** [comment]
bot, build com.github.afrantzis.Bless

**Comment by bilelmoussaoui** [comment]
> I've added an upscaled version of the icon, but upstream's email bounces, so it's hard to contact them about Bless at the moment.

Why not just report an issue on the Github project?

File: com.github.afrantzis.Bless.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>com.github.afrantzis.Bless</id>
```
Comment (bilelmoussaoui) [inline com.github.afrantzis.Bless.metainfo.xml:3]: It would be nice to upstream the metainfo

**Comment by bilelmoussaoui** [comment]
The app crashes over here whenever I try to open a file 
```
Marshaling activate signal
Exception in Gtk# callback delegate
  Note: Applications can use GLib.ExceptionManager.UnhandledException to handle the exception.
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x0001c] in <7c320154f5854083bbf670d5ed1705eb>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x00019] in <7c320154f5854083bbf670d5ed1705eb>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0001d] in <7c320154f5854083bbf670d5ed1705eb>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy+<>c.<StartDispatching>b__9_0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <7c320154f5854083bbf670d5ed1705eb>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x00184] in <7c320154f5854083bbf670d5ed1705eb>:0 
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <1544d2d88bf049059a891378c137b939>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <1544d2d88bf049059a891378c137b939>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <1544d2d88bf049059a891378c137b939>:0 
  at System.Delegate.DynamicInvokeImpl (System.Object[] args) [0x000e7] in <1544d2d88bf049059a891378c137b939>:0 
  at System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) [0x00008] in <1544d2d88bf049059a891378c137b939>:0 
  at System.Delegate.DynamicInvoke (System.Object[] args) [0x00000] in <1544d2d88bf049059a891378c137b939>:0 
  at GLib.Signal.ClosureInvokedCB (System.Object o, GLib.ClosureInvokedArgs args) [0x0007d] in <ed39f21b9e9343dcbd442a17ad356a9f>:0 
  at GLib.Signal+SignalClosure.Invoke (GLib.ClosureInvokedArgs args) [0x0000c] in <ed39f21b9e9343dcbd442a17ad356a9f>:0 
  at GLib.Signal+SignalClosure.MarshalCallback (System.IntPtr raw_closure, GLib.Value* return_val, System.UInt32 n_param_vals, GLib.Value* param_values, System.IntPtr invocation_hint, System.IntPtr marshal_data) [0x00050] in <ed39f21b9e9343dcbd442a17ad356a9f>:0 
  at GLib.ExceptionManager.RaiseUnhandledException (System.Exception e, System.Boolean is_terminal) [0x00000] in <ed39f21b9e9343dcbd442a17ad356a9f>:0 
  at GLib.Signal+SignalClosure.MarshalCallback (System.IntPtr raw_closure, GLib.Value* return_val, System.UInt32 n_param_vals, GLib.Value* param_values, System.IntPtr invocation_hint, System.IntPtr marshal_data) [0x00000] in <ed39f21b9e9343dcbd442a17ad356a9f>:0 
  at Gtk.Application.gtk_main () [0x00000] in <7aab76e87bce48a4b45cf7fa613cb70c>:0 
  at Gtk.Application.Run () [0x00000] in <7aab76e87bce48a4b45cf7fa613cb70c>:0 
  at BlessMain..ctor (System.String[] args) [0x00000] in <6ea14f55f4cf4e83b3e2404f35eaca1e>:0 
  at BlessMain.Main (System.String[] args) [0x00000] in <6ea14f55f4cf4e83b3e2404f35eaca1e>:0 

```

**Comment by FakeShemp** [comment]
@bilelmoussaoui I think this is a mono runtime issue. Found some details about it here: https://github.com/mono/mono/issues/12557

A workaround was mentioned in that thread, so I implemented it in this manifest for the time being.

**Comment by FakeShemp** [comment]
bot, build com.github.afrantzis.Bless

**Comment by barthalion** [comment]
Have you contacted upstream author about your submission?

**Comment by FakeShemp** [comment]
I've sent an email with no answer, but I also made an issue on Github right now.

https://github.com/afrantzis/bless/issues/29

**Comment by FakeShemp** [comment]
bot, build com.github.afrantzis.Bless

**Comment by barthalion** [comment]
bot, build com.github.afrantzis.Bless

**Comment by barthalion** [comment]
Hey @afrantzis, I'll invite you to Flathub repository just in case you want to change something/merge a pull request yourself. Let me know if you have any questions.

/merge @afrantzis

**Pull request closed without merging**